### PR TITLE
feat: add delegate endpoints

### PR DESCRIPTION
**Pull request by iamacook**
## Overview

This adds the add, edit, delete and delete (all?) endpoints to the SDK as well as their types. There are still a few TODOs:

- It seems as though the `DELETE` method is not allowed by the CORS policy. As such, the delete functions aren't working and I can't confirm the return types (although the typed ones are taken from [the docs](https://safe-global.github.io/safe-client-gateway/docs/routes/delegates/index.html)).
- The get/add functions have a union return type for the `'/v1/chains/{chainId}/delegates'` endpoint that needs to be fixed.

### CODE REVIEW

**Comment by mikhailxyz** [comment]
> It seems as though the DELETE method is not allowed by the CORS policy. As such, the delete functions aren't working and I can't confirm the return types (although the typed ones are taken from [the docs](https://safe-global.github.io/safe-client-gateway/docs/routes/delegates/index.html)).

Does the backend enforce it on purpose, or did they forget to enable it?

**Comment by iamacook** [comment]
> Does the backend enforce it on purpose, or did they forget to enable it?

I haven't consulted them yet. This is a draft that we're going to come back to.

**Comment by DiogoSoaress** [comment]
_note:_ last commit was to try the HTTP `DELETE` method for https://github.com/safe-global/safe-client-gateway/blob/08655807c78b81f79cb4be91a1bc95f95d09ca37/src/routes/delegates/routes.rs#L36-L49 and https://github.com/safe-global/safe-client-gateway/blob/08655807c78b81f79cb4be91a1bc95f95d09ca37/src/routes/delegates/routes.rs#L51-L71

File: src/types/api.ts [hunk 1]
```diff
@@ -144,6 +153,46 @@ export interface paths {
...
+        get: operations['get_delegates']
+        parameters: {
+          path: {
+            chainId: string
+          }
+          query: DelegatesRequest
+        }
+      }
+    | {
+        get: operations['add_delegate']
```
Comment (katspaugh) [inline src/types/api.ts:167]: `get` is the HTTP method. It should be `get` and `post` in this case.

File: src/utils.ts [hunk 2]
```diff
@@ -66,3 +66,36 @@ export async function fetchData<T>(url: string, body?: unknown): Promise<T> {
 
   return json
 }
+
+export async function deleteData<T>(url: string, body?: unknown): Promise<T> {
```
Comment (katspaugh) [inline src/utils.ts:70]: Why a separate function? It looks 90% the same as the fetchData.

Comment (iamacook) [reply src/utils.ts:70]: Piggyback on this, `fetch` is now [available in Node](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V18.md#2022-04-19-version-1800-current-bethgriggs).

**Comment by iamacook** [comment]
I'm going to close this. We can use this as reference if we revisit it later.

**Pull request closed without merging**